### ISSUE TITLE
Newsletter subscription dialog autofocuses email field, skipping past name

### ISSUE THREAD

**Issue by WestonThayer** [AssistivLabs/ghost-demo#3]
### Issue Summary

https://github.com/AssistivLabs/ghost-demo/assets/709153/c9b32b20-9f54-475d-abe4-0f45da181388

**Expected:** When subscribing to a newsletter, the modal/dialog that opens should focus the first visible form field, which is Name.

**Actual:** The second visible field (Email) is focused, which is a 2.4.3 failure (focus order is not logical).

## Notes

The issue is higher severity because, for example, a blind screen reader user might assume Email is the only field, fill it out, and submit the form. There is no NVDA announcement for them, although a sighted user is able to view the error on the name field.

**WCAG 2.2 AA:** fails [2.4.3 Focus Order (Level A)](https://www.w3.org/TR/WCAG22/#focus-order) and later on [4.1.3 Status Messages (Level AA)](https://www.w3.org/TR/WCAG22/#status-messages).

**Test file:** https://github.com/assistivlabs-external/ghost-demo/blob/main/tests/001-subscribe-to-newsletter.spec.mjs

**Associated Deploy**: [11:12 AM PST May 1 2024](https://github.com/foo)

**PRs in Deploy:**

- https://github.com/AssistivLabs/ghost-demo/pull/5 @JamieWay 

### Steps to Reproduce

1. Using [assistivlabs](https://assistivlabs.com) with NVDA and Chrome, go to the landing page for a newsletter, signed out
2. Using your keyboard, tab to the **Subscribe** link
3. Press Space
4. Observe which form field is focused

### Ghost Version

canary

### Node.js Version

18.x

### How did you install Ghost?

source

### Database type

MySQL 5.7

### Browser & OS version

Chrome 123 & Windows 10

### Code of Conduct

- [X] I agree to be friendly and polite to people in this repository
